src/index.js

```diff
@@ -15,8 +16,11 @@ async function run() {
 		manifestPath: validator.getManifestPath()
 	};
 	const uploader = new UploadCourseContent(inputs, api);
+	const rewriter = new LinkRewriter(inputs.contentDirectory, api);
 
-	await uploader.uploadCourseContent(validator.getInstanceDomain(), validator.getCourseOrgUnitId());
+	const manifest = await uploader.uploadCourseContent(validator.getInstanceDomain(), validator.getCourseOrgUnitId());
+
+	await rewriter.rewriteLinks(validator.getInstanceDomain(), validator.getCourseOrgUnitId(), manifest);
```

Comment:
for consideration - it looks like uploadCourseContent and rewriteLinks are both updating the content.  Are you worried about how non-atomic this is?   Someone could be using the course while this runs, and we're updating old content with links that are known to be broken, and then updating them later.  If there's an error then the whole course could be left in a state where links are broken.  I personally think it's cleaner if the upload course content step only creates new objects and builds the manifest map, and all updates are leftt o the rewrite links step.  That would be more atomic, so each existing object will be left in either the old or new state and never an in-between state.   I don't think it's much work to fix, but I'd be happy creating an issue if you don't think it's worth fixing at this point.

Reply:
I think I'd opt for an issue at this point.
Unless this workflow makes sense:
- We're comfortable creating the content with empty HTML documents on the first pass. (For file topics the file is mandatory :S)
- Update the content based on created manifest
- Update all the content with the updated HTML

Reply:
I think I am fine with that, but I'll create an issue anyway.  It can be addressed in a future PR.